roadmap/implementers-guide/src/types/messages.md

```diff
@@ -26,3 +29,34 @@ struct UpwardMessage {
 	pub data: Vec<u8>,
 }
 ```
+
+## Horizontal Message
+
+This is a message sent from a parachain to another parachain that travels through the relay chain.
+This message ends up in the recipient's mailbox. A size of a horizontal message is defined by its
+`data` payload.
+
+```rust,ignore
+struct HorizontalMessage {
+	/// The para that will get this message in its downward message queue.
+	pub recipient: ParaId,
+	/// The message payload.
+	pub data: Vec<u8>,
+}
+```
+
+## Downward Message
+
+A message that go down from the relay chain to a parachain. Such a message could be initiated either
+as a result of an operation took place on the relay chain or sent using a horizontal message.
+
+```rust,ignore
+enum DownwardMessage {
+	/// Some funds were transferred into the parachain's account. The hash is the identifier that
+	/// was given with the transfer.
+	TransferInto(AccountId, Balance, Remark),
+	/// This downward message is a result of a horizontal message represented as opaque bytes sent
+	/// by the specified sender.
+	HorizontalMessage(AccountId, Vec<u8>),
```

Comment:
Should it be `ParaId` instead of `AccountId`? Maybe even not that.

Reply:
Ah, yeah, I ofc meant `ParaId`. Changed it to that.

Why do you think it might be not `ParaId`? I assumed only other paras can send a horizontal message?

Reply:
Yeah I thought we might embed the sender in the message but it seems better to keep the `ParaId` in there.